### DISCUSSION TITLE
How to "granularize" commits and pulls/merges to individual files?

### DISCUSSION THREAD

**Discussion by proofer** [jesseduffield/lazygit#2226] [Q&A]
Not too knowledgeable about git. Lazygit (& git?) seems oriented to batching of files. But I like to have one commit message per file. Similarly when pulling from a remote I like to separately evaluate each file's merge. What's my suggested Lazygit workflow where there are 2+ files involved?

**Comment by mark2185** [comment]
The same workflow as you would have with `git`, you can either stage and commit one file per commit, or write commit messages with multiple lines.

As for pulling from a remote, you'll have only one merge commit (by default), you can't really split that up. You could pull without a merge commit and then manually commit that as usual if you really want to.

But look into `git rebase`, that might suit you better.
